src/session_impl.cpp

```diff
@@ -1685,8 +1691,13 @@ namespace aux {
 					!= m_settings.get_str(settings_pack::listen_interfaces));
 
 		apply_pack(&pack, m_settings, this);
+
 		m_disk_thread.set_settings(&pack, m_alerts);
 
+#if TORRENT_USE_INVARIANT_CHECKS
+		check_invariant();
```

Comment:
some session_handle methods call the apply settings method. And many session imply set methods call the impl.

Do you want to have this call removed and added to the 10 caller functions?